core/src/callback.jl

```diff
@@ -289,18 +289,23 @@ function discrete_control_affect!(integrator, compound_variable_idx)
     # Get the discrete_control node to which this compound variable belongs
     discrete_control_node_id = discrete_control.node_id[compound_variable_idx]
 
-    # Get the indices of all conditions that this control node listens to
-    where_node_id = searchsorted(discrete_control.node_id, discrete_control_node_id)
-
-    # Get the truth state for this discrete_control node
-    truth_values = cat(
-        [
-            [ifelse(b, "T", "F") for b in discrete_control.condition_value[i]] for
-            i in where_node_id
-        ]...;
-        dims = 1,
-    )
-    truth_state = join(truth_values, "")
+    # Get the first index of the (compound) variables that this control node listens to
+    variable_idx = searchsortedfirst(discrete_control.node_id, discrete_control_node_id)
+    truth_value_idx = 1
+
+    # Build up the truth state from the truth values per (compound) variable per greater than
+    while variable_idx <= length(discrete_control.node_id) &&
```

Comment:
I think it's much more readable to for loop over these nodes, instead of jumping with a searchsortedfirst and having a while loop?